src/controllers/authController.js

```diff
@@ -0,0 +1,154 @@
+        try{
+         if(!email || !password){
+            return res.json("email and password are required")
+        }
+         const checkEmail = `SELECT * FROM users WHERE email=$1`
+         const value = [email];
+          const returnedEmail = await pool.query(checkEmail, value);
+            if(!returnedEmail.rows[0]){
+                return res.status(400).json({
+                status: "bad request",
+                message: "incorrect email or password"
+                })
+            }else if(returnedEmail.rows[0]['suspend_status'] === 'FALSE'){
+                return res.status(400).json({
+                    status: "bad request",
+                    message: "user has been suspended"
+                })
+            }
+        const match = await bcrypt.compare(password, returnedEmail.rows[0].password);
+        if(match) {
+         jwt.sign({email, password} , process.env.AUTHKEY, {expiresIn : '30d'} , (err, token)=>{
+            if(err){
+                console.log(err);
+            }else{
+                return res.status(200).json({
+                    status: "ok",
+                    message: "signed in successfully",
+                    id : returnedEmail.rows[0]["userid"],
+                    email: returnedEmail.rows[0]["email"],
+                    first_name: returnedEmail.rows[0]["first_name"],
+                    last_name: returnedEmail.rows[0]["last_name"],
+                    phonenumber: returnedEmail.rows[0]["phonenumber"],
+                    createdat: returnedEmail.rows[0]["createdat"],
+                    email_verified: returnedEmail.rows[0]["email_verified"],
+                    gender_id: returnedEmail.rows[0]["gender_id"],
+                    admin: returnedEmail.rows[0]["admin"],
+                    auth_provider: returnedEmail.rows[0]["auth_provider"],
+                    suspend_status: returnedEmail.rows[0]["suspend_status"],
+                    pro: returnedEmail.rows[0]["pro"],
+                    token : token
+                })
+            }
+     })
+        }else{
+         res.status(400).json({status: "bad request", message: "incorrect email or password"})
+     }
+     
+    }catch(err){
+            console.log(err)
+        }
+    }
+
+
+
+
+    /****
+     * @static
+     *
+     * @param {object} request - {email, first_name, last_name, phonenumber, category, admin, password} -> The request payload sent to the controller
+     * @param {object} response
+     *
+     * @returns {object} - status, message, id, email, first_name, last_name, phonenumber, category, createdat, auth_provider
+     *
+     * @description Sign up users
+     * @memberOf Authentication
+     *
+     *****/
+
+   static async  signUp(req, res) {
+       const email = req.body.email;
+       const first_name = req.body.first_name;
+       const last_name = req.body.last_name;
+       const phonenumber = req.body.phonenumber;
+       const category = req.body.category;
```

Comment:
Category is not in the current dB for users

Reply:
Done !